compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -4617,6 +4667,9 @@ private ParserRuleContext getNextRuleForTypeDescriptor() {
             case TYPE_DESC_IN_PATH_PARAM:
                 endContext();
                 return ParserRuleContext.PATH_PARAM_ELLIPSIS;
+            case TYPE_DESC_BEFORE_IDENTIFIER_IN_GROUPING_KEY:
+                endContext(); // End TYPE_DESC_BEFORE_IDENTIFIER_IN_GROUPING_KEY
+                return ParserRuleContext.BINDING_PATTERN_STARTING_IDENTIFIER;
```

Comment:
Since we are starting a context for `BINDING_PATTERN_STARTING_IDENTIFIER` we can use that to distinguish the two `variable-name` cases. So I think we don't need `GROUPING_KEY_LIST_VAR_DECLARATION` can u check

Reply:
Need this to identify the alternative paths for  `expressions` in group by

Reply:
We end `BINDING_PATTERN_STARTING_IDENTIFIER` context after resolving `variable-name` but there are instances where we need to identify that we are in a `GROUPING_KEY_LIST_VAR_DECLARATION` context where `BINDING_PATTERN_STARTING_IDENTIFIER` doesn't exist in recovering instances and also to distinguish valid expression alternatives way  after `BINDING_PATTERN_STARTING_IDENTIFIER` has ended.